### ISSUE TITLE
Allow more flexible legend placement also for lineage legend

### ISSUE THREAD

**Issue by Marius1311** [theislab/cellrank#496]
- [ ] Additional function parameters, change functionality or change defaults?
- [ ] New estimator in `cr.tl.estimators`?
- [ ] New kernel in `cr.tl.kernels`?
- [ ] New gene trend model in `cr.ul.models`?
- [x] New plotting function in `cr.pl`?
- [ ] Other?

When plotting gene expression trends, I can place the `cell_color` legend very flexibly, using e.g. `top right out`. However, the same is not possible for the lineage legend, i.e. running the below code
```
model = cr.ul.models.GAM(adata_raw, n_knots=7)

fig_kwargs = {'model': model,
              'data_key': 'magic_imputed_data', 
              'genes': ['Pdx1'], 
              'time_key': 'palantir_pseudotime', 
              'weight_threshold': (0.05, 0.05),
              'cell_color': 'clusters',
              'legend_loc': 'top left out',
              'obs_legend_loc': 'top right out',
              'same_plot': True
             }
cr.pl.gene_trends(adata_raw, **fig_kwargs)
```
gives a warning
```
MatplotlibDeprecationWarning: Unrecognized location 'top left out'. Falling back on 'best'; valid locations are
	best
	upper right
	upper left
	lower left
	lower right
	right
	center left
	center right
	lower center
	upper center
	center
This will raise an exception in 3.3.
```
and shows 
![image](https://user-images.githubusercontent.com/24494650/108040113-33d75c80-703d-11eb-85b2-e316153cd5ba.png)

To unify the behaviour of both legends, I would allow passing the same location keywords.

**Comment by michalk8** [comment]
Sure, sounds reasonable (we could do it for other plots as well).

**Comment by Marius1311** [comment]
I think it's most relevant to plotting functions that support 2 legends (PAGA and gene trend plotting, and probably some others I forgot) but it won't hurt for the others as well as it will be more consistent.

**Comment by michalk8** [comment]
closed via #487

**Comment by Marius1311** [comment]
Fantastic, thanks.
